### PR TITLE
Fix #10821. Allow page reloads, navigating away and back etc.

### LINKED ISSUE #10821: LivePreview is closed if browser is reloaded by user
Steps to reproduce:

1. Open GettingStarted project (or any other previewable project).
2. Start LivePreview.
3. In the browser window that is opened, press the reload button.

Result: LivePreview session is closed.
Expected: LivePreview reloads the document being previewed.

### PR DESCRIPTION
**Pull request by busykai**
_Fixes multi-browser case only_.

Fix #10821: wait for 5 seconds after the last browser has closed connection before shutting down entire LiveDevelopment session.

It's a very simple and robust fix to address page reloads inflicted by `protocol.reload()` or by user-driven page reloads (or navigating back and forth on the page links).

A more sophisticated fix would consist of the remote end (a browser) letting know that the page is being unloaded. However, the only case where it would actually help is when the page is reloaded by the protocol command. It still would be impossible to distinguish intentional navigating away from a reload or navigating away by clicking a link on the page being previewed.

CC: @redmunds, @sebaslv

### CODE REVIEW

**Comment by busykai** [comment]
Plus this PR has a couple of clean-ups.

File: src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js [hunk 1]
```diff
@@ -272,7 +272,6 @@
         },
         
         onClose: function () {
-            // TODO: This is absolutely temporary solution.
```
Comment (le717) [inline src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js]: Just curious, I guess this not an "absolutely temporary solution" after all?

**Comment by le717** [comment]
Thanks for putting this up. Hopefully it can be merged soon. I've been hitting this case a lot recently.

File: src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js (hunk 1 above)
Comment (busykai) [inline src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js]: @le717 :) it's been reworked from its original version, but the comment was left. what we have now seems to be satisfactory.

File: src/LiveDevelopment/LiveDevMultiBrowser.js [hunk 2]
```diff
@@ -557,9 +557,13 @@ define(function (require, exports, module) {
                     .on("ConnectionClose.livedev", function (event, msg) {
                         // close session when the last connection was closed
                         if (_protocol.getConnectionIds().length === 0) {
-                            if (exports.status <= STATUS_ACTIVE) {
-                                _close(false, "detached_target_closed");
-                            }
+                            setTimeout(function () {
+                                if (_protocol.getConnectionIds().length === 0) {
```
Comment (MarcelGerber) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: Can't we combine the two conditions to `if (_protocol.getConnectionIds().length === 0 && exports.status <= STATUS_ACTIVE)`?

Comment (busykai) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: I did it intentionally (for some reason), but you're right, no need.

**Comment by busykai** [comment]
@MarcelGerber, did you try it? it's ready to merged otherwise, it seems.

**Comment by MarcelGerber** [comment]
Thanks.

**Pull request merged**